### DISCUSSION TITLE
Data accuracy

### DISCUSSION THREAD

**Discussion by JRascagneres** [gekkekoe/esphome-ecodan-hp#98] [Announcements]
Hey all,

I know the electricity data isn't overly accurate in the ecodan so I have a separate shelly setup. Do we know if the heat / production data is accurate? Planning to use that and my shelly to come up with a more accurate COP.

Thanks!

**Comment by gekkekoe** [comment]
From what I've read its quite useable. Within 2-3%. I'm experimenting with the daily energy sensor to sum the estimated output power during the day, and it's quite matching the heatpump numbers, it needs a bit more testing before it will be released.

you can connect an external pulse meter to the ecodan and then the input reading are accurate as well.

**Reply by JRascagneres** [reply]
Sounds good thanks! I'm already using the shelly and the daily outputted numbers from the heat pump. Was just wondering how accurate.

 I've slapped an integral and then utility meter on the estimated calculation and it seems closeish too - so I can get a 'live' reading before the end of the day.

**Reply by gekkekoe** [reply]
It has been pushed, give it a try, it's around 0-3% difference (keep in mind that it starts from 0, so probably next whole day will have okay numbers)

**Reply by JRascagneres** [reply]
Got it. Will give it a go. I already have my own sensor setup for it but better for it to be on-device.

**Reply by DamijanD** [reply]
Mine started with: -35,930,777,275,334,656.000 kWh :-)

**Reply by gekkekoe** [reply]
yeah, will be fixed after midnight :)

**Reply by JRascagneres** [reply]
It seems decent but about 10% off for me. Good for the live estimate though.

**Reply by gekkekoe** [reply]
wait and compare with totals with heatpump next midnight, for me it's about 3% diff from the heatpump

**Reply by JRascagneres** [reply]
Okay will give it another day. I'm wondering its because mine cycles a bit? Here is todays:

<img width="1441" alt="image" src="https://github.com/user-attachments/assets/a6037129-91d7-43a7-81ab-ac1c87bef0ea" />

Then this is my sensor adding DHW and Heating output:
<img width="1454" alt="image" src="https://github.com/user-attachments/assets/06edd52b-bacb-46c1-98a3-26d26bb43c52" />

Its 16.97 from the estimate and 19.33 from heat pump.


It does line up with operations: 
<img width="1452" alt="image" src="https://github.com/user-attachments/assets/52e19f49-1faa-4435-b95b-fed0f2d47d8a" />

Not sure what the difference is but I think its okay.

**Reply by gekkekoe** [reply]
Ah so for defrost, I deduct that amount since we are extracting that heat. Every start of a run introduces some noise, so a lot of cycling can make it diff more I guess.

Also dhw is included, I have not split it. So for me stats for yesterday

heatpump:
DWH 8.04 kWh
heating 112.30 kWh

Estimation was 116.62 kWh, I had 3 defrost.
Each defrost is approx 0,4 kWh.

Without defrost correction:  112.3+8.04 vs 116.62 is 3%

With defrost correction:  112.3+8.04 vs 116.62 + 3*0,4 is 2.1%

I think if you add DHW, then the error will be much lower. Keep in mind that if you use a utility meter on the estimated power output, it does also include the DHW heat generation

**Reply by JRascagneres** [reply]
Yeah, I think its the start / stop of a run thats introducing 'noise'. and given I'm only outputting either 16.97 from the estimate or the 19.33 it may be enough noise to make the difference. Yours are lot larger numbers so any noise may be a smaller %age. 

I also have a couple defrosts in there. I think for now I'm going to stick with the heat pump data for long term records but the estimate seems good for daily 'live' data. This lines up very close with my own integral + utility meter I made based on the live estimate.

**Reply by gekkekoe** [reply]
yesterday
estimation 118.68 kWh
heatpump 117.5 + 1.65 kWh.
It's really close when there are no defrost/cycling. Only one start/stop from the DHW run.
